### PR TITLE
Optimize export

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

**Comment by itsmontoya** [comment]
[Optimize MrT exports](https://trello.com/c/cYpen0KZ/301-optimize-mrt-exports)

File: exporter.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	m  *MrT
+	w  io.Writer
+	hw *shasher.HashWriter
+	mf *Match
+}
+
+func (e *exporter) exportFrom(f file.Interface, s *seeker.Seeker) (err error) {
+	var ltid string
+	if ltid, err = peekLastTxn(s); err == nil && ltid == e.txnID {
```
Comment (shahzilabid) [inline exporter.go:31]: Where do we check for this err not being nil?

File: mrT.go [hunk 2]
```diff
@@ -564,36 +572,96 @@ func (m *MrT) Import(r io.Reader, fn ForEachFn) (lastTxn string, err error) {
...
+
+func (m *MrT) exportArchive(e *exporter) (err error) {
+	var f *file.File
+	if f, err = file.Open(path.Join(m.dir, "archive", m.name+".tdb")); err != nil {
+		return
+	}
+	defer f.Close()
+
+	as := seeker.New(f)
+	defer as.SetFile(nil)
```
Comment (shahzilabid) [inline mrT.go]: Can we get a new line after this defer? Feels off

**Review by shahzilabid — COMMENTED** [review]
Couple comments. Also, pretty sure the answer is no but just wanted to make sure.. this PR wouldn't change the internal structure of the db right? Seems like it's just touching export

**Comment by itsmontoya** [comment]
Correct, this only affects exports. The underlying structure of the system is untouched.

File: exporter.go (hunk 1 above)
Comment (itsmontoya) [reply exporter.go:31]: So if err is not nil, that means we have zero transactions. I could also ensure ltid is not empty, but I figured checking for lack of error was more correct.

Comment (shahzilabid) [reply exporter.go:31]: Got it

**Review by shahzilabid — APPROVED** [review]
Good shit as always!

**Review by OneOfOne — APPROVED** [review]
Great work brother

**Pull request merged**